app/controllers/posts_controller.rb

```diff
@@ -210,6 +210,13 @@ def update
       edit_reason: params[:post][:edit_reason]
     }
 
+    # Check if there is no edit conflict caused by two users saving different
+    # versions of the post.
+    raw_old = params[:post][:raw_old]
+    if !raw_old.empty? && raw_old != post.raw
+      return render json: { errors: [I18n.t('edit_conflict')] }, status: 409
```

Comment:
There should be a spec for that.

Reply:
We have a pattern for rendering json errors, you can have a look at `render_json_error`